model-optimizer/extensions/back/ReverseInputChannels.py

```diff
@@ -94,8 +94,23 @@ class ReverseChannelsPropagationDown(BackReplacementPattern):
 
         'Shape': lambda node, rc: ReverseChannelsPropagationDown.pass_rc_through_shape(node, rc),
         'ShapeOf': lambda node, rc: ReverseChannelsPropagationDown.pass_rc_through_shape(node, rc),
+
+        'Pad': lambda node, rc: ReverseChannelsPropagationDown.pass_rc_through(node, rc),
     }
 
+    @staticmethod
+    def pass_rc_through(node: Node, reverse_channels: Node):
+        # detaching reverse_channels node from the graph
+        reverse_channels.out_port(0).get_connection().set_source(
+            reverse_channels.in_port(0).get_connection().get_source())
+        reverse_channels.in_port(0).disconnect()
+
+        node.out_port(0).get_connection().set_source(reverse_channels.out_port(0))
+        node.out_port(0).disconnect()
+        node.out_port(0).connect(reverse_channels.in_port(0))
+
+        return True
```

Comment:
functionally not clear about why a graph `ReverseChannels->Pad` is not functionally equivalent to `Pad->ReverseChannels`